web/js/codeworld_shared.js

```diff
@@ -260,42 +287,48 @@ function registerStandardHints(successFunc)
         // Filter out CallStack constraints.
         line = line.replace(/:: HasCallStack =>/g, '::');
 
-        var wordStart = 0;
-        if (line.startsWith("type ") || line.startsWith("data ")) {
-            wordStart += 5;
-
-            // Hide kind annotations.
-            var kindIndex = line.indexOf(" ::");
-            if (kindIndex != -1) {
-                line = line.substr(0, kindIndex);
+        if (line.startsWith("-- ")) {
+            doc += line.replace(/(\-\- \| |\-\-   )/g, "") + "<br>";
```

Comment:
I don't believe you want the `<br>` there.  I believe you just want `\n`.  Line wrapping in haddock comments is not significant except in preformatted blocks.

Reply:
Work with `\n`, replacing.